feature/experimental/gribi/ate_tests/hierarchical_gribi_entries/ack_in_presence_other_routes/route_ack_test.go

```diff
@@ -0,0 +1,293 @@
+		WithMin("1.0.0.0").
+		WithMax("1.0.0.254").
+		WithCount(250)
+
+	flow := ate.Traffic().NewFlow("Flow").
+		WithSrcEndpoints(srcEndPoint).
+		WithDstEndpoints(dstEndPoint).
+		WithHeaders(ethHeader, ipv4Header)
+
+	ate.Traffic().Start(t, flow)
+	time.Sleep(15 * time.Second)
+	ate.Traffic().Stop(t)
+
+	time.Sleep(time.Minute)
+
+	flowPath := ate.Telemetry().Flow(flow.Name())
+	if got := flowPath.LossPct().Get(t); got > 0 {
+		t.Errorf("LossPct for flow %s got %g, want 0", flow.Name(), got)
+	}
+}
+
+// testArgs holds the objects needed by a test case.
+type testArgs struct {
+	ctx     context.Context
+	clientA *gribi.Client
+	dut     *ondatra.DUTDevice
+	ate     *ondatra.ATEDevice
+	top     *ondatra.ATETopology
+}
+
+//Configure network instance
+func configureNetworkInstance(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	d := &telemetry.Device{}
+	ni1 := d.GetOrCreateNetworkInstance(instance)
+	ni1.Type = telemetry.E_NetworkInstanceTypes_NETWORK_INSTANCE_TYPE(*ygot.Int64(1))
+
+	dutConfPath := dut.Config().NetworkInstance(instance)
+	dutConfPath.Replace(t, ni1)
+}
+
+// configStaticRoute configures a static route.
+func configStaticRoute(t *testing.T, dut *ondatra.DUTDevice, prefix string, nexthop string) *telemetry.NetworkInstance_Protocol_Static {
+	d := &telemetry.Device{}
+	ni1 := d.GetOrCreateNetworkInstance(instance)
+	ni1.Enabled = ygot.Bool(true)
+	ni1.Name = ygot.String(instance)
+	ni1.Description = ygot.String("Static route added by gNMI-OC")
+	static := ni1.GetOrCreateProtocol(telemetry.PolicyTypes_INSTALL_PROTOCOL_TYPE_STATIC, "static-1")
+	static.Enabled = ygot.Bool(true)
+	static.Identifier = telemetry.E_PolicyTypes_INSTALL_PROTOCOL_TYPE(*ygot.Int64(10))
+	sr := static.GetOrCreateStatic(prefix)
+	nh := sr.GetOrCreateNextHop("nhg1")
+	nh.NextHop = fpoc.UnionString(nexthop)
+
+	return sr
+}
+
+// routeAck configures a IPv4 entry through clientA. Ensure that the entry via ClientA
+// is active through AFT Telemetry.
+func routeAck(ctx context.Context, t *testing.T, args *testArgs) {
+	// Add an IPv4Entry for 1.0.0.0/8 pointing to ATE port-3 via gRIBI-A,
+	// ensure that the entry is active through AFT telemetry
+	t.Logf("Add an IPv4Entry for %s pointing to ATE port-3 via gRIBI-A", ateDstNetCIDR)
+	args.clientA.AddNH(t, nhIndex, atePort3.IPv4, instance, fluent.InstalledInRIB)
+	args.clientA.AddNHG(t, nhgIndex, map[uint64]uint64{nhIndex: 1}, instance, fluent.InstalledInRIB)
+	args.clientA.AddIPv4(t, ateDstNetCIDR, nhgIndex, instance, "", fluent.InstalledInRIB)
+
+	// Verify the entry for 1.0.0.0/8 is active through AFT Telemetry.
+	ipv4Path := args.dut.Telemetry().NetworkInstance(instance).Afts().Ipv4Entry(ateDstNetCIDR)
+	if got, want := ipv4Path.Prefix().Get(t), ateDstNetCIDR; got != want {
+		t.Errorf("ipv4-entry/state/prefix got %s, want %s", got, want)
+	}
+	// Verify the entry for 1.0.0.0/8 is active through Traffic.
+	srcEndPoint := args.top.Interfaces()[atePort1.Name]
+	dstEndPoint := args.top.Interfaces()[atePort2.Name]
```

Comment:
Should be:

```
dstEndPoint := args.top.Interfaces()[atePort3.Name]
```

Reply:
Thanks Likai.  per my understanding, static route is preferred over gRIBI route, hence traffic sent to ATE port-2.

Reply:
cc: @xw-g 

the intention was that we define an admin distance for gRIBI, such that it is able to fit into the standard means of determining which the preferred protocol route is. originally, the test plan assumed gRIBI to be more preferable than static. I think per [this update](https://github.com/openconfig/gribi/pull/66/files) that Xiao made, we've declared that the preference should be static-over-gRIBI.

Likely we should update the test plan to:
 * fix the bug you highlight about send/receive on the same port
 * check that the default static admin distance is more preferable than gRIBI
 * check that a higher admin distance static is less preferable than gRIBI

this would feel reasonable to cover the admin distance part of the specification. the static admin distance can be specified with the `metric` leaf for a particular static route next-hop [docs](http://ops.openconfig.net/branches/models/master/docs/openconfig-network-instance.html#network-instances-network-instance-protocols-protocol-static-routes-static-next-hops-next-hop-config-metric).

Reply:
I agree with @robshakir 's suggestion. However, one comment: 

@robshakir, I feel like the mentioned [docs](http://ops.openconfig.net/branches/models/master/docs/openconfig-network-instance.html#network-instances-network-instance-protocols-protocol-static-routes-static-next-hops-next-hop-config-metric) is about metric, not about AD (Administrative Distance, or Route Preference), especially given the wording `When multiple next-hops are specified for a static route, the metric is utilised to determine`. @dplore can comment from OC's perspective? 

@robshakir @dplore I also did a quick check of current OC tree for static route, seems like there is no option for user to change the AD of static route. Can you help double confirm? 

If my understanding is correct, then probably we need to 
1. Setup a BGP between DUT-ATE. 
2. Or, wait for OC to augmentation?

Reply:
@xw-g explained to me in a sync that network operators sometimes have to use static routes in an emergency situation to correct a faulty routing choice made by a dynamic routing protocol, and that's why static routes have a higher preference than BGP or ISIS. And gRIBI is acting as a client that represents SDN as a dynamic routing agent, so routes coming from gRIBI will have a fixed AD that is lower than static routes, but higher than BGP and ISIS.

And static routes configured by either OpenConfig or CLI will have the same AD.

Given this, I agree that let's update TE-3.6 to prefer static route. @azubaidhar

Reply:
@xw-g the metric here is basically on a `<prefix,next-hop>` tuple, such that you can map to something like:

```
ip route 2.2.2.2/32 1.1.1.1 240
ip route 2.2.2.2/32 3.3.3.3 10
````

where in these implementations the metric used here is also interpreted as admin distances. 

today, we don't have something to change the admin distance for a protocol - we should probably look to add this.